src/screens/Documents/View/index.js

```diff
@@ -159,8 +160,11 @@ class DocumentsViewScreen extends NavigatorComponent<
   onDocumentSubmit = () => {
     const { openedDocumentId } = this.props;
     if (openedDocumentId == null) return;
-
+    this.setState({ loading: true });
     this.props.uploadDocument(openedDocumentId);
+    setTimeout(() => {
+      this.setState({ loading: false });
+    }, 3000);
   };
```

Comment:
A hard-coded 3000ms timer is better than nothing but still not the best solution. But we'll do it this way for now. .